### ISSUE TITLE
Errors with py-hunspell 0.5.0

### ISSUE THREAD

**Issue by thierry-FreeBSD** [mike-fabian/ibus-typing-booster#23]
Hello,

All tests pass when py-hunspell 0.4.1 is installed (Python 3.6).

But with py-hunspell 0.5.0, they produce many errors:

```
Traceback (most recent call last):
  File "../engine/hunspell_table.py", line 717, in _update_candidates
    pp_phrase=self._pp_phrase)
  File "../engine/tabsqlitedb.py", line 480, in select_words
    x for x in self.hunspell_obj.suggest(input_phrase)])
  File "../engine/hunspell_suggest.py", line 318, in suggest
    dictionary.encoding, 'replace'))
  File "../engine/hunspell_suggest.py", line 315, in <listcomp>
    for x in
AttributeError: 'str' object has no attribute 'decode'
tabsqlitedb.select_words() input_phrase=b'diff' p_phrase=b'' pp_phrase=b''
```

**Comment by mike-fabian** [comment]
There seem to be problems with pyhunspell 0.5.0:

https://github.com/blatinier/pyhunspell/issues/32

As long as these are not resolved, I am afraid I  cannot do anything, if even this doesn’t work

https://github.com/blatinier/pyhunspell/issues/32#issuecomment-326904426

then there is nothing I can do.

As soon as this issue is fixed in pyhunspell, I’ll see what I can do to make pyhunspell >= 0.5.0
work in ibus-typing-booster.

**Comment by thierry-FreeBSD** [comment]
Same error with py-hunspell 0.5.3:

```
tabsqlitedb.select_words() input_phrase=b'diff' p_phrase=b'' pp_phrase=b''
Hunspell.suggest() input_phrase=b'diff'
Traceback (most recent call last):
  File "../engine/hunspell_table.py", line 717, in _update_candidates
    pp_phrase=self._pp_phrase)
  File "../engine/tabsqlitedb.py", line 480, in select_words
    x for x in self.hunspell_obj.suggest(input_phrase)])
  File "../engine/hunspell_suggest.py", line 318, in suggest
    dictionary.encoding, 'replace'))
  File "../engine/hunspell_suggest.py", line 315, in <listcomp>
    for x in
AttributeError: 'str' object has no attribute 'decode'
```

**Comment by thierry-FreeBSD** [comment]
According to https://github.com/blatinier/pyhunspell/issues/32 this problem should have been fixed with pyhunspell 0.5.4, but I'm still seeing the same error (with ibus-typing-booster 1.5.35):

```
process_key_event() KeyEvent object: val=102 code=0 state=0x00000000 name='f' unicode='f' msymbol='f' control=False mod1=False mod5=False release=False
_insert_string_at_cursor() string_to_insert=['f']
_insert_string_at_cursor() self._typed_string=['d', 'i', 'f']
_insert_string_at_cursor() self._typed_string_cursor=3
_update_transliterated_strings() self._typed_string=['d', 'i', 'f', 'f']
_update_transliterated_strings() self._transliterated_strings={'NoIme': 'diff', 't-latn-post': 'diff'}
_update_candidates() self._typed_string=['d', 'i', 'f', 'f']
tabsqlitedb.select_words() input_phrase=b'diff' p_phrase=b'' pp_phrase=b''
Hunspell.suggest() input_phrase=b'diff'
Traceback (most recent call last):
  File "../engine/hunspell_table.py", line 717, in _update_candidates
    pp_phrase=self._pp_phrase)
  File "../engine/tabsqlitedb.py", line 480, in select_words
    x for x in self.hunspell_obj.suggest(input_phrase)])
  File "../engine/hunspell_suggest.py", line 318, in suggest
    dictionary.encoding, 'replace'))
  File "../engine/hunspell_suggest.py", line 315, in <listcomp>
    for x in
AttributeError: 'str' object has no attribute 'decode'
```

**Comment by mike-fabian** [comment]
You are right, pyhunspell 0.5.4 returns Unicode now. I have to adapt ibus-typing-booster
a bit for that:

diff --git a/engine/hunspell_suggest.py b/engine/hunspell_suggest.py
index ddb6d15..22f6267 100755
--- a/engine/hunspell_suggest.py
+++ b/engine/hunspell_suggest.py
@@ -310,8 +310,7 @@ class Hunspell:
                             suggested_words[input_phrase] = 0
                         extra_suggestions = [
                             unicodedata.normalize(
-                                itb_util.NORMALIZATION_FORM_INTERNAL, x.decode(
-                                    dictionary.encoding))
+                                itb_util.NORMALIZATION_FORM_INTERNAL, x)
                             for x in
                             dictionary.pyhunspell_object.suggest(
                                 input_phrase_nfc.encode(

**Comment by mike-fabian** [comment]
I’ll release a new  version soon.

**Comment by mike-fabian** [comment]
Fixed in 1.5.36: https://github.com/mike-fabian/ibus-typing-booster/releases/tag/1.5.36
